### PR TITLE
Add a new column for the mutation position in the sub-peptide

### LINKED ISSUE #176: Location of mutation in sequence
Hi there, 

Is there any chance of having a column showing the location of the mutated AA (I guess only for missense)?

Best,
Nils

### PR DESCRIPTION
**Pull request by susannasiebert**
This PR includes an updated algorithm to find the corresponding wt epitope sequence for a mt epitope. Previously we used a brute force approach and compared all wt epitope sequences to the mt epitope and simply picked the one with the most number of matching amino acids. Now we use the indel length to find the positions of wt epitopes to compare.

I also added another test for frameshifts.

Closes #176

### CODE REVIEW

File: pvacseq/lib/parse_output.py [hunk 1]
```diff
@@ -15,63 +15,186 @@
...
+
+def find_mutation_position_for_insertions(wt_epitope_seq, mt_epitope_seq):
+    while len(wt_epitope_seq) > 0:
+        wt_epitope_seq = wt_epitope_seq[1:]
+        mt_epitope_seq = mt_epitope_seq[:-1]
+        for i,(wt_aa,mt_aa) in enumerate(zip(wt_epitope_seq,mt_epitope_seq)):
+            if wt_aa != mt_aa:
+                mutation_position = i
+                break
+        if mutation_position != 0:
```
Comment (tmooney) [inline pvacseq/lib/parse_output.py]: Is it possible for `mutation_position` to not get set?  Or for it to never be non-zero?

Comment (susannasiebert) [inline pvacseq/lib/parse_output.py]: Good question. Now that you pointed it out I realized that this whole subroutine isn't doing the right thing at all. I pushed a commit to fix this bug.

**Comment by acoffman** [comment]
+1

File: pvacseq/lib/parse_output.py [hunk 2]
```diff
@@ -15,63 +15,160 @@
...
+    for key in sorted(iedb_results.keys(), key = lambda x: int(x.split('|')[-1])):
+        result = iedb_results[key]
         (wt_iedb_result_key, mt_position) = key.split('|', 1)
+        wt_results = wt_iedb_results[wt_iedb_result_key]
         if result['variant_type'] == 'missense':
-            iedb_results[key]['wt_epitope_seq'] = wt_iedb_results[wt_iedb_result_key][mt_position]['wt_epitope_seq']
-            iedb_results[key]['wt_scores']      = wt_iedb_results[wt_iedb_result_key][mt_position]['wt_scores']
-        else:
-            wt_results        = wt_iedb_results[wt_iedb_result_key]
+            iedb_results[key]['wt_epitope_seq']    = wt_results[mt_position]['wt_epitope_seq']
```
Comment (tmooney) [inline pvacseq/lib/parse_output.py]: Could this be simplified to `result['wt_epitope_seq'] = ...`?  (Similar question for subsequent `iedb_results[key]` within this loop.)

Comment (susannasiebert) [inline pvacseq/lib/parse_output.py]: That's an interesting point. I'm not sure if `result` is a pointer or a copy. I think python handles these as pointers so it should work. I will test when I get back.

**Comment by susannasiebert** [comment]
I agree that the matching algorithm is pretty complicated. I'm open to suggestions on how to make this code more readable. I could split it into more subroutines. Also, I tried to document the matching for inframe indels with comments. If you think those are helpful I can add similar comments to the rest of the cases.

**Comment by susannasiebert** [comment]
@tmooney I decided to break the `match_wildtype_and_mutant_entries` function into smaller functions. I also added more comments.

**Pull request merged**